code/modules/mob/living/basic/lavaland/watcher/watcher_overwatch.dm

```diff
@@ -0,0 +1,158 @@
+/**
+ * Automatically shoot at a target if they do anything while this is active on them.
+ */
+/datum/action/cooldown/mob_cooldown/watcher_overwatch
+	name = "Overwatch"
+	desc = "Keep a close eye on the target's actions, automatically firing upon them if they act."
+	button_icon = 'icons/mob/actions/actions_ecult.dmi'
+	button_icon_state = "eye"
+	background_icon_state = "bg_demon"
+	overlay_icon_state = "bg_demon_border"
+	cooldown_time = 20 SECONDS
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_INCAPACITATED
+	click_to_activate = TRUE
+	shared_cooldown = null
+	/// Furthest range we can activate ability at
+	var/max_range = 7
+	/// Type of projectile to fire
+	var/projectile_type = /obj/projectile/temp/watcher
+	/// Sound the projectile we fire makes
+	var/projectile_sound = 'sound/weapons/pierce.ogg'
+	/// Time to watch for
+	var/overwatch_duration = 3 SECONDS
+
+/datum/action/cooldown/mob_cooldown/watcher_overwatch/New(Target, original)
+	. = ..()
+	melee_cooldown_time = overwatch_duration
+
+/datum/action/cooldown/mob_cooldown/watcher_overwatch/PreActivate(atom/target)
+	if (target == owner)
+		return
+	if (ismecha(target))
+		var/obj/vehicle/sealed/mecha/mech = target
+		var/list/drivers = mech.return_drivers()
+		if (!length(drivers))
+			return
+		target = drivers[1]
+	if (!isliving(target))
+		return
+	if (get_dist(owner, target) > max_range)
+		return
+	return ..()
+
+/datum/action/cooldown/mob_cooldown/watcher_overwatch/Activate(mob/living/target)
+	var/mob/living/living_owner = owner
```

Comment:
where's the anti-spam 5 minutes dummy cooldown?

Reply:
I understand the use of this for the gaze attack which uses a do_after, but spam-clicking this ability will just toggle targetting it on and off and clicking on anything with it on will use the ability. Not sure what the dummy cooldown would be achieving.

Reply:
it's more about keeping it symmetrical with the other activates imo